services/core/WeatherAgent/weather/weather.py

```diff
@@ -0,0 +1,503 @@
+# information, apparatus, product, software, or process disclosed, or
+# represents that its use would not infringe privately owned rights.
+#
+# Reference herein to any specific commercial product, process, or
+# service by trade name, trademark, manufacturer, or otherwise does not
+# necessarily constitute or imply its endorsement, recommendation, or
+# favoring by the United States Government or any agency thereof, or
+# Battelle Memorial Institute. The views and opinions of authors
+# expressed herein do not necessarily state or reflect those of the
+# United States Government or any agency thereof.
+#
+# PACIFIC NORTHWEST NATIONAL LABORATORY
+# operated by BATTELLE for the UNITED STATES DEPARTMENT OF ENERGY
+# under Contract DE-AC05-76RL01830
+
+# }}}
+import os
+import sys
+import logging
+import json
+import time
+import urllib2
+import pytz
+from dateutil import parser
+from datetime import datetime, timedelta
+
+from volttron.platform.vip.agent import Agent, Core, PubSub, RPC, compat
+from volttron.platform.agent import utils
+from volttron.platform.agent.utils import (get_aware_utc_now,
+                                           format_timestamp)
+from volttron.platform.messaging import headers as headers_mod, topics
+
+__version__ = '2.0'
+
+HEADER_NAME_DATE = headers_mod.DATE
+HEADER_NAME_CONTENT_TYPE = headers_mod.CONTENT_TYPE
+
+utils.setup_logging()
+_log = logging.getLogger(__name__)
+
+location = ["local_tz_long", "observation_location", "display_location",
+            "station_id"]
+time_topics = ["local_time_rfc822", "local_tz_short", "local_tz_offset",
+               "local_epoch", "observation_time", "observation_time_rfc822",
+               "observation_epoch"]
+temperature = ["temperature_string", "temp_f", "temp_c", "feelslike_c",
+               "feelslike_f", "feelslike_string", "windchill_c",
+               "windchill_f", "windchill_string", "heat_index_c",
+               "heat_index_f", "heat_index_string"]
+wind = ["wind_gust_kph", "wind_string", "wind_mph", "wind_dir",
+        "wind_degrees", "wind_kph", "wind_gust_mph", "pressure_in"]
+cloud_cover = ["weather", "solarradiation", "visibility_mi", "visibility_km",
+               "UV"]
+precipitation = ["dewpoint_string", "precip_today_string", "dewpoint_f",
+                 "dewpoint_c", "precip_today_metric", "precip_today_in",
+                 "precip_1hr_in", "precip_1hr_metric", "precip_1hr_string"]
+pressure_humidity = ["pressure_trend", "pressure_mb", "relative_humidity"]
+observation_epoch = 'observation_epoch'
+
+all_current = [observation_epoch] + temperature + wind + cloud_cover + precipitation + pressure_humidity
+
+history_points = ["tempm", "tempi", "dewptm", "dewpti", "hum", "wspdm", "wspdi", "wgustm", "wgusti", "wdird", "wdire", "vism", "visi", "pressurem", "pressurei", "windchillm", "windchilli", "heatindexm", "heatindexi", "precipm", "precipi", "conds", "icon", "fog", "rain", "snow", "hail", "thunder", "tornado", "metar"]
```

Comment:
what is a metar?

Reply:
METAR is also known as Meteorological Terminal Aviation Routine Weather Report, is the most common format in the world for the transmission of observational weather data.

Reply:
hmm learn somehting new every day :)